### PR TITLE
Degeneral UI feedback

### PR DESCRIPTION
**Pull request by MeanBoyCousin**
# Task: Options chain UI upgrades based on feedback from @degeneral-defi 

## Description

- Darker green on the buying column for better readability.
- Now showing a maximum of six expiry dates instead of four.
- Modals now autofocus the order size input field.
- Switched to using form components to allow actions to be executed on enter key press.
- Modals can now be closed by escape key press or clicking the lightbox.
- DHV exposure column is now visible by default.
- Better ARIA and titles added where missing.
- User can now quickly switch between buying and selling where available.
- Removes the options trading context and hoists all state into the global context.
- Now showing price impact for order size on buying and selling.
- Option chain UI now shows prices as premium only.
- Updates the date format on the chain to DDMMMYY.
- Shortens the height of the chain rows to the minimum accessible height.
- Now showing ATM via a black dashed line and some green triangles.
- Hovering a row now highlights the whole row green for better visibility across flavours.
- Added Joyride tutorials to the buying & selling modals.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update
- [ ] Quality of life improvement
- [ ] Package upgrades

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have NATSPEC'd any new functions
- [x] If appropriate, I have properly tested my new functionality

### CODE REVIEW

**Comment by degeneral-defi** [comment]
- (row highlighting) Mousing over low strike calls highlights lime on calls and grey on puts. However mousing over high strike calls puts the puts in lime and the calls in grey, even when the mouse is hovering over the call side. I would expect that which ever side is moused over (call/put) is the one that is in lime.

**Review by thundering-silence — APPROVED** [review]
gg

File: packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      run={buyTutorialIndex !== undefined}
+      showProgress
+      showSkipButton
+      spotlightPadding={8}
+      stepIndex={buyTutorialIndex}
+      steps={[
+        {
+          disableBeacon: true,
+          target: "#buy-symbol",
+          content: `Here you can see the asset you have selected to buy. This is represented as "underlying asset - expiry date - strike price - option flavor".`,
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx:32]:
```suggestion
          content: `Here you can see the option you have selected to buy. This is represented as "underlying asset - expiry date - strike price - option type (put or call)".`,
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        },
+        {
+          target: "#buy-price-per-option",
+          content:
+            "These prices represent the cost per option. The premium and fees are displayed separately for transparency, and you can also see how the price is impacted based on order size.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx:37]:
```suggestion
            "These prices represent the total amount you pay for the option. The premium and fees are displayed separately for transparency, and you can also see how the premium is impacted by slippage based on order size.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        },
+        {
+          target: "#buy-balance",
+          content:
+            "This price represents the estimated remaining balance of the USDC in your wallet after the transaction.",
+        },
+        {
+          target: "#buy-input",
+          content:
+            "You can use this input to specify how many contracts you wish to buy. We allow denominations as small as 0.01.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx:52]:
```suggestion
            "You can use this input to specify how many contracts you wish to buy. You can trade denominations as small as 0.01.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        },
+        {
+          target: "#buy-button",
+          content:
+            "You can click here to approve your spending and complete your purchase. If you do not already have your wallet connected, this button can handle that. Transactions will require two confirmations to be considered complete, and the interface will automatically close once your transaction is complete.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Buy/index.tsx:57]:
```suggestion
            "You can click here to approve your spending and complete your buy trade. If you do not already have your wallet connected, this button can handle that. Transactions will require two confirmations to be considered complete, and the interface will automatically close once your transaction is complete.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      run={sellTutorialIndex !== undefined}
+      showProgress
+      showSkipButton
+      spotlightPadding={8}
+      stepIndex={sellTutorialIndex}
+      steps={[
+        {
+          disableBeacon: true,
+          target: "#sell-symbol",
+          content: `Here you can see the asset you have selected to buy. This is represented as "underlying asset - expiry date - strike price - option flavor".`,
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:32]:
```suggestion
          content: `Here you can see the option you have selected to sell. This is represented as "underlying asset - expiry date - strike price - option type (put or call)".`,
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+          target: "#sell-collateral",
+          content:
+            "These filters allow you to specify the type of collateral you wish to use and how much you will add.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:37]:
```suggestion
            "To sell an option you will be required to post collateral. These filters allow you to specify the type of collateral you wish to use and how much you will add.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+          target: "#sell-price-per-option",
+          content:
+            "These prices represent the cost per option. The premium and fees are displayed separately for transparency, and you can also see how the price is impacted based on order size.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:42]:
```suggestion
            "These prices represent the total amount you receive per option. The premium and fees are displayed separately for transparency, and you can also see how the premium is impacted by slippage based on order size.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+          target: "#sell-collateral-required",
+          content:
+            "This price represents the total amount of collateral you must provide to complete the transaction.",
+        },
+        {
+          target: "#sell-total-price",
+          content:
+            "This price represents the total amount in USDC you will receive after fees.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:52]:
```suggestion
            "This price represents the total amount in USDC you will receive as premium after fees.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+          target: "#sell-balances",
+          content:
+            "These prices represent the estimated remaining balance of the USDC and WETH in your wallet after the transaction.",
+        },
+        {
+          target: "#sell-input",
+          content:
+            "You can use this input to specify how many contracts you wish to sell. We allow denominations as small as 0.01.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:62]:
```suggestion
            "You can use this input to specify how many contracts you wish to sell. You can sell denominations as small as 0.01.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+          target: "#sell-button",
+          content:
+            "You can click here to approve your spending and complete your purchase. If you do not already have your wallet connected, this button can handle that. Transactions will require two confirmations to be considered complete, and the interface will automatically close once your transaction is complete.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:67]:
```suggestion
            "You can click here to approve your spending for collateral required and complete your sell. If you do not already have your wallet connected, this button can handle that. Transactions will require two confirmations to be considered complete, and the interface will automatically close once your transaction is complete.",
```

File: packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+          target: "#quick-switch",
+          content:
+            "This handy button will allow you to quickly toggle between buying and selling for the specified strike price and option flavor.",
```
Comment (ugolino) [inline packages/front-end/src/components/optionsTrading/Tutorials/Sell/index.tsx:72]:
```suggestion
            "This handy button will allow you to quickly toggle between buying and selling for the selected option.",
```

**Review by ugolino — APPROVED** [review]

**Pull request merged**